src/main/kotlin/com/karumi/maxibonkata/KarumiHQs.kt

```diff
@@ -0,0 +1,33 @@
+package com.karumi.maxibonkata
+
+class KarumiHQs(val chat: Chat = ConsoleChat()) {
+    var maxibonsLeft: Int = 10
+
+    fun openFridge(developer: Developer) {
+        grabMaxibons(developer)
+        if (shouldBuyMoreMaxibons()) {
+            notifyWeShouldBuyMaxibon(developer)
+            buyMaxibons()
+        }
+    }
+
+    fun openFridge(developers: List<Developer>) {
+        developers.forEach { developer ->
+            openFridge(developer)
+        }
+    }
+
+    private fun grabMaxibons(developer: Developer) {
+        maxibonsLeft -= developer.maxibonsToGrab
+        maxibonsLeft = Math.max(maxibonsLeft, 0)
+    }
+
+    private fun shouldBuyMoreMaxibons(): Boolean = maxibonsLeft <= 2
```

Comment:
It's not so critical, but `magic numbers` could be replaced by constants.

Reply:
Agree, I don't have strong feelings with this specific case because it is only used once, though.